nexus/src/internal_api/params.rs

```diff
@@ -8,14 +8,15 @@ use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 use std::fmt;
 use std::net::SocketAddr;
+use std::net::SocketAddrV6;
 use std::str::FromStr;
 use uuid::Uuid;
 
 /// Sent by a sled agent on startup to Nexus to request further instruction
 #[derive(Serialize, Deserialize, JsonSchema)]
 pub struct SledAgentStartupInfo {
     /// the address of the sled agent's API endpoint
-    pub sa_address: SocketAddr,
+    pub sa_address: SocketAddrV6,
```

Comment:
One thing I don't quite understand: I expected this to trigger a change to the generated OpenAPI spec, and a corresponding change in the sled agents (simulated and real) to use a v6 address.  But it looks like the generated type for this represents this field with a string:
https://github.com/oxidecomputer/omicron/blob/b7f7ea6592734940a765afc5538864fd0d08e472/sled-agent/src/sim/server.rs#L76

so the distinction between v4+v6 and v6-only is not there.  I gather that's because json-schema says the schema for an IP address is just a string:
https://docs.rs/schemars/latest/src/schemars/json_schema_impls/primitives.rs.html#54

Kind of a bummer...but well beyond the scope of this PR.

Reply:
Yeah it's definitely a bummer. It seems that a large number of types in the generated clients are serialized as a string, and require that explicit `.to_string()` call. I'm not sure why, since it seems like the serialization should happen after the type is constructed. But yeah, beyond the scope of this PR. Is there an issue or something you'd like me to create for this?

Reply:
I don't think so.  Thanks.